engine/src/widget.lcb

```diff
@@ -1260,6 +1273,42 @@ end syntax
 
 //
 
+/**
+Summary: Returns the current widget.
+Returns: A widget object.
+
+Example:
+
+-- In a widget
+private variable mSelf as Widget
+public handler OnCreate() returns nothing
+	-- Keep a reference to this widget
+	put this widget into mSelf
+
+	-- defined in separate module library
+	SetEventCallback(EventCallback)
+end handler
+
+-- may be called from another module library
+private handler EventCallback() returns nothing
+	-- update internal variables
+
+	-- notify ide of changes
+	trigger all in mSelf
+end handler
+
+Description:
+This widget evaluates to the current widget. This can be used to retain a reference for occasions where widget handlers may be called from another module, where the current widget may not be valid.
```

Comment:
@livecodeian could you hard wrap this at 80 chars. Also wondering if it would be good to be a bit more explicit about the difference between `my script object` and `this widget` as it could be confusing for users. I think that a description of the context where it is valid would be good here also. Or perhaps a description of where it is invalid may be easier e.g. asynchronous callbacks.